### DISCUSSION TITLE
Docker image for artillery with playwright

### DISCUSSION THREAD

**Discussion by jhengy** [artilleryio/artillery#2512] [Artillery]
## Context
Hi, I'm following the guide from https://www.artillery.io/docs/docker to run artillery test in docker, with the intention of eventually using the released docker image `artilleryio/artillery:latest` to [run artillery test on Gitlab CI/CD](https://www.artillery.io/docs/cicd/gitlab-cicd). 

## Ask
However, it seems that the image does not come bundled with dependencies needed by playwright. May i know if there is another public image released by the team that comes bundled with playwright? Or is it that we have to build our own docker image(something similar to https://github.com/artilleryio/artillery/blob/main/packages/artillery-engine-playwright/Dockerfile)?

**Comment by vikramjb** [comment]
@jhengy Playwright is one of the engines Artillery uses so it I doble it will get bundled as a default package in artillery dockerfile. However the docker file you have provided seems to have the items you are looking for. Wouldn't that be more useful for you?

**Comment by hassy** [comment]
Hi @jhengy 👋 What's the use case? We don't recommend running Artillery + Playwright tests directly in CI (it won't scale) but to trigger them from CI to [run on AWS Fargate instead](https://www.artillery.io/docs/load-testing-at-scale/aws-fargate).

If you would still rather execute Playwright directly on CI, you'll need to install Playwright as a dependency or use a custom Docker image that bundles Playwright.

**Reply by jhengy** [reply]
The plan is indeed to run Artillery + Playwright tests  on AWS Fargate using Gitlab CI/CD. Does it mean that there is a difference between `artillery run` vs `artillery run-fargate` in a sense that `artillery run` requires the host environment to have playwright dependencies installed whereas `artillery run-fargate` will help us installing the necessary playwright dependencies remotely in Fargate containers?

**Reply by bernardobridge** [reply]
Yes @jhengy, exactly. `run-fargate` offloads the actual running of the tests (i.e. VU execution) to the Fargate containers, so the process within the Gitlab container will only do things like metrics aggregation, etc. 

For that reason the workers (Fargate containers) contain all the dependencies needed to run the Playwright tests, which is one of the reasons we recommend using that. The other reason is that you can configure memory and cpu requirements, and scale the test horizontally. Which for Playwright tests (resource-heavy) is important.

**Reply by MSauchanka** [reply]
Could you please recommend Dokerfile that will work to run UI performance with Playwright through Artillery. I cannot use Artillery GHA actions to run it on fargate/lambda because action's author is not marked as verified. That's my company security limitation.
Will this work https://github.com/artilleryio/artillery/blob/main/packages/artillery-engine-playwright/Dockerfile ?
